tests/unit/export/models/validationManager.spec.ts

```diff
@@ -0,0 +1,389 @@
+      //This uses the logical assignment operator (||=), which assigns a value only if the left-hand side is falsy.
+      (updatedCallbackParameters.exportInputParams.callbackUrls ||= []).push(addedCallbackUrl[0]);
+
+      nock(jobManagerURL)
+        .get('/jobs')
+        .query(completedExportParams as Record<string, string>)
+        .reply(200, [])
+        .persist();
+      nock(jobManagerURL)
+        .get('/jobs')
+        .query(inProgressExportParams as Record<string, string>)
+        .reply(200, inProgressJobsResponse);
+      nock(jobManagerURL)
+        .get(`/jobs/${inProgressJobsResponse[0].id}`)
+        .query({ shouldReturnTasks: false })
+        .reply(200, inProgressJobsResponse[0])
+        .persist();
+      nock(jobManagerURL).get(`/jobs/${inProgressJobsResponse[1].id}`).query({ shouldReturnTasks: false }).reply(200, inProgressJobsResponse[1]);
+      nock(jobManagerURL)
+        .get('/jobs')
+        .query(pendingExportParams as Record<string, string>)
+        .reply(200, []);
+
+      nock(jobManagerURL)
+        .put(`/jobs/${inProgressJobsResponse[0].id}`, JSON.stringify({ parameters: updatedCallbackParameters }))
+        .reply(200, []);
+
+      const result = await validationManager.checkForExportDuplicate(productId, version, catalogId, roi, crs, addedCallbackUrl);
+
+      expect(result).toEqual(processingResponse);
+    });
+
+    it('should return an processing export job and add a callback', async () => {
+      const { crs, productId, version, catalogId, roi } = dupParams;
+      const matchingJob = [{ ...inProgressJobsResponse[0] }];
+      // Perform a deep copy of the parameters object
+      const updatedCallbackParameters = JSON.parse(JSON.stringify(matchingJob[0].parameters)) as ExportJobParameters;
+      // Use type assertion to safely delete the property
+      delete (matchingJob[0].parameters.exportInputParams as { callbackUrls?: unknown }).callbackUrls;
+      updatedCallbackParameters.exportInputParams.callbackUrls = addedCallbackUrl;
+
+      nock(jobManagerURL)
+        .get('/jobs')
+        .query(completedExportParams as Record<string, string>)
+        .reply(200, [])
+        .persist();
+      nock(jobManagerURL)
+        .get('/jobs')
+        .query(inProgressExportParams as Record<string, string>)
+        .reply(200, matchingJob);
+      nock(jobManagerURL).get(`/jobs/${matchingJob[0].id}`).query({ shouldReturnTasks: false }).reply(200, matchingJob[0]).persist();
+      nock(jobManagerURL)
+        .get('/jobs')
+        .query(pendingExportParams as Record<string, string>)
+        .reply(200, []);
+
+      nock(jobManagerURL)
+        .put(`/jobs/${matchingJob[0].id}`, JSON.stringify({ parameters: updatedCallbackParameters }))
+        .reply(200, []);
+
+      const result = await validationManager.checkForExportDuplicate(productId, version, catalogId, roi, crs, addedCallbackUrl);
+
+      expect(result).toEqual(processingResponse);
```

Comment:
check also for `updateExportCallbackURLs()` has been called once with new callbacks

Reply:
uncomment the added expectation

Reply:
commented on purpose to not forget. there is an issue with the haveBennCalledWith params. Lets talk in person on Tuesday